panic: assertion failed: tx closed in storage pkg test

2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 4
2015/09/01 15:16:00 store.index: compact 5
2015/09/01 15:16:00 store.index: compact 6
2015/09/01 15:16:00 store.index: compact 7
2015/09/01 15:16:00 store.index: compact 8
2015/09/01 15:16:00 store.index: compact 9
2015/09/01 15:16:00 store.index: compact 10
2015/09/01 15:16:00 store.index: compact 11
2015/09/01 15:16:00 store.index: compact 12
2015/09/01 15:16:00 store.index: compact 13
2015/09/01 15:16:00 store.index: compact 14
2015/09/01 15:16:00 store.index: compact 15
2015/09/01 15:16:00 store.index: compact 16
2015/09/01 15:16:00 store.index: compact 17
2015/09/01 15:16:00 store.index: compact 18
2015/09/01 15:16:00 store.index: compact 19
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 4
2015/09/01 15:16:00 store.index: compact 5
2015/09/01 15:16:00 store.index: compact 6
2015/09/01 15:16:00 store.index: compact 7
2015/09/01 15:16:00 store.index: compact 8
2015/09/01 15:16:00 store.index: compact 9
2015/09/01 15:16:00 store.index: compact 10
2015/09/01 15:16:00 store.index: compact 11
2015/09/01 15:16:00 store.index: compact 12
2015/09/01 15:16:00 store.index: compact 13
2015/09/01 15:16:00 store.index: compact 14
2015/09/01 15:16:00 store.index: compact 15
2015/09/01 15:16:00 store.index: compact 16
2015/09/01 15:16:00 store.index: compact 17
2015/09/01 15:16:00 store.index: compact 18
2015/09/01 15:16:00 store.index: compact 19
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 0
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 0
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 storage: resume scheduled compaction at 1
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 storage: resume scheduled compaction at 2
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 4
2015/09/01 15:16:00 store.index: compact 5
2015/09/01 15:16:00 store.index: compact 6
2015/09/01 15:16:00 store.index: compact 7
2015/09/01 15:16:00 store.index: compact 8
2015/09/01 15:16:00 store.index: compact 9
2015/09/01 15:16:00 store.index: compact 10
2015/09/01 15:16:00 store.index: compact 11
2015/09/01 15:16:00 store.index: compact 12
2015/09/01 15:16:00 store.index: compact 13
2015/09/01 15:16:00 store.index: compact 14
2015/09/01 15:16:00 store.index: compact 15
2015/09/01 15:16:00 store.index: compact 16
2015/09/01 15:16:00 store.index: compact 17
2015/09/01 15:16:00 store.index: compact 18
2015/09/01 15:16:00 store.index: compact 19
2015/09/01 15:16:00 store.index: compact 1
2015/09/01 15:16:00 store.index: compact 2
2015/09/01 15:16:00 store.index: compact 3
2015/09/01 15:16:00 store.index: compact 4
2015/09/01 15:16:00 store.index: compact 5
2015/09/01 15:16:00 store.index: compact 6
2015/09/01 15:16:00 store.index: compact 7
2015/09/01 15:16:00 store.index: compact 8
2015/09/01 15:16:00 store.index: compact 9
2015/09/01 15:16:00 store.index: compact 10
2015/09/01 15:16:00 store.index: compact 11
2015/09/01 15:16:00 store.index: compact 12
2015/09/01 15:16:00 store.index: compact 13
2015/09/01 15:16:00 store.index: compact 14
2015/09/01 15:16:00 store.index: compact 15
2015/09/01 15:16:00 store.index: compact 16
2015/09/01 15:16:00 store.index: compact 17
2015/09/01 15:16:00 store.index: compact 18
2015/09/01 15:16:00 store.index: compact 19
2015/09/01 15:16:01 store.index: compact 3
2015/09/01 15:16:01 store.index: compact 3
panic: assertion failed: tx closed
goroutine 184 [running]:
github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt._assert(0xc8203d0a00, 0xa1de30, 0x9, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt/db.go:782 +0xfc
github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt.(*Cursor).seek(0xc82003bc58, 0xc672ea, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt/cursor.go:148 +0xf7
github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt.(*Bucket).Bucket(0xc8203d0aa8, 0xc672ea, 0x3, 0x3, 0x462514)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt/bucket.go:111 +0x2ac
github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt.(*Tx).Bucket(0xc8203d0a90, 0xc672ea, 0x3, 0x3, 0x0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/Godeps/_workspace/src/github.com/boltdb/bolt/tx.go:91 +0x59
github.com/coreos/etcd/storage/backend.(*batchTx).UnsafePut(0xc820402820, 0xc672ea, 0x3, 0x3, 0xc820402b80, 0x11, 0x11, 0xc820317ce0, 0xd, 0xd)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/batch_tx.go:38 +0x8e
github.com/coreos/etcd/storage.(*store).put(0xc82009d300, 0xc82037cad0, 0x3, 0x8, 0x0, 0x0, 0x0, 0x1)
	github.com/coreos/etcd/storage/_test/_obj_test/kvstore.go:439 +0x641
github.com/coreos/etcd/storage.(*store).Put(0xc82009d300, 0xc82037cad0, 0x3, 0x8, 0x0, 0x0, 0x0, 0x0)
	github.com/coreos/etcd/storage/_test/_obj_test/kvstore.go:77 +0xc5
github.com/coreos/etcd/storage.TestKVTxnBlockNonTnxOperations.func2()
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/kv_test.go:414 +0xa1
github.com/coreos/etcd/storage.TestKVTxnBlockNonTnxOperations.func4(0xc82007c0f8, 0xc820370f60)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/kv_test.go:421 +0x3c
created by github.com/coreos/etcd/storage.TestKVTxnBlockNonTnxOperations
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/kv_test.go:423 +0x430
goroutine 1 [chan receive]:
testing.RunTests(0xae1630, 0xc858c0, 0x2a, 0x2a, 0x957001)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc82004df00, 0xc88f20)
	/tmp/workdir/go/src/testing/testing.go:494 +0xe5
main.main()
	github.com/coreos/etcd/storage/_test/_testmain.go:196 +0x385
goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/tmp/workdir/go/src/runtime/asm_amd64.s:1696 +0x1
goroutine 118 [select]:
github.com/coreos/etcd/storage/backend.(*backend).run(0xc8202fdd00)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/backend.go:80 +0x243
created by github.com/coreos/etcd/storage/backend.New
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/backend.go:48 +0x4ea
goroutine 121 [select]:
github.com/coreos/etcd/storage/backend.(*backend).run(0xc82037a8c0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/backend.go:80 +0x243
created by github.com/coreos/etcd/storage/backend.New
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/backend.go:48 +0x4ea
goroutine 160 [runnable]:
github.com/coreos/etcd/storage/backend.(*backend).Close(0xc8203e5e80, 0x0, 0x0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/backend/backend.go:92 +0x90
github.com/coreos/etcd/storage.(*store).Close(0xc82009d300, 0x0, 0x0)
	github.com/coreos/etcd/storage/_test/_obj_test/kvstore.go:314 +0xc9
github.com/coreos/etcd/storage.cleanup(0x7fe5573151f0, 0xc82009d300, 0xc820069b00, 0x28)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/kv_test.go:695 +0x35
runtime.Goexit()
	/tmp/workdir/go/src/runtime/panic.go:320 +0x101
testing.(*common).FailNow(0xc82046e3f0)
	/tmp/workdir/go/src/testing/testing.go:336 +0x57
testing.(*common).Fatalf(0xc82046e3f0, 0xa6de60, 0x25, 0xc82003ae58, 0x1, 0x1)
	/tmp/workdir/go/src/testing/testing.go:379 +0x95
github.com/coreos/etcd/storage.TestKVTxnBlockNonTnxOperations(0xc82046e3f0)
	/home/travis/gopath/src/github.com/coreos/etcd/gopath/src/github.com/coreos/etcd/storage/kv_test.go:434 +0x7ed
testing.tRunner(0xc82046e3f0, 0xc85ba8)
	/tmp/workdir/go/src/testing/testing.go:456 +0xdd
created by testing.RunTests
	/tmp/workdir/go/src/testing/testing.go:561 +0xaa4
FAIL	github.com/coreos/etcd/storage	1.604s

--- a/storage/kv_test.go
+++ b/storage/kv_test.go
@@ -430,7 +430,7 @@ func TestKVTxnBlockNonTnxOperations(t *testing.T) {
 		s.TxnEnd(id)
 		select {
 		case <-done:
-		case <-time.After(10 * time.Millisecond):
+		case <-time.After(100 * time.Millisecond):
 			t.Fatalf("#%d: operation failed to be unblocked", i)
 		}
 	}

